### PR TITLE
refactor(clang-analyzer): remove dead code

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/strings.c [hunk 1]
```diff
@@ -1542,7 +1542,6 @@ int vim_vsnprintf_typval(char *str, size_t str_m, const char *fmt, va_list ap_st
         p++;
         if (length_modifier == 'l' && *p == 'l') {
           // double l = long long
-          length_modifier = 'L';
```
Comment (zeertzjq) [inline src/nvim/strings.c:1545]: I don't think this should be removed, otherwise the code will be quite confusing.

**Comment by zeertzjq** [comment]
Also, all these changes are applicable to Vim.

File: src/nvim/strings.c (hunk 1 above)
Comment (dundargoc) [reply src/nvim/strings.c:1545]: Unsure if clang-analyzer has an ignore directive, will check.

**Comment by dundargoc** [comment]
I think the windows errors are real O_o

File: src/nvim/strings.c (hunk 1 above)
Comment (dundargoc) [reply src/nvim/strings.c]: This section is absolutely needed, for Windows.

**Comment by dundargoc** [comment]
Superseded by https://github.com/neovim/neovim/pull/25531

**Pull request closed without merging**